File: app/src/main/java/com/zibete/proyecto1/ui/ResizableImageViewProfile.java
package com.zibete.proyecto1.ui;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.util.AttributeSet;

import androidx.appcompat.widget.AppCompatImageView;

public class ResizableImageViewProfile extends AppCompatImageView {

    public ResizableImageViewProfile(Context context) {
        super(context);
        init();
    }

    public ResizableImageViewProfile(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    public ResizableImageViewProfile(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    private void init() {
        // Le decimos que mantenga proporciones automáticamente
        setAdjustViewBounds(true);
        setScaleType(ScaleType.FIT_CENTER);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        Drawable drawable = getDrawable();

        if (drawable == null) {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec);
            return;
        }

        int width = MeasureSpec.getSize(widthMeasureSpec);
        int height;

        // Mantener relación de aspecto (ancho / alto)
        float imageRatio = (float) drawable.getIntrinsicWidth() / (float) drawable.getIntrinsicHeight();

        if (width > 0) {
            height = (int) (width / imageRatio);
        } else {
            height = drawable.getIntrinsicHeight();
        }

        setMeasuredDimension(width, height);
    }
}

File: app/src/main/java/com/zibete/proyecto1/ui/ResizableImageViewProfile.kt
package com.zibete.proyecto1.ui

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatImageView

class ResizableImageViewProfile @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : AppCompatImageView(context, attrs, defStyleAttr) {

    init {
        adjustViewBounds = true
        scaleType = ScaleType.FIT_CENTER
    }

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val d = drawable

        // Si no hay imagen, medir normal
        if (d == null) {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec)
            return
        }

        val intrinsicW = d.intrinsicWidth
        val intrinsicH = d.intrinsicHeight

        // Caso raro: drawables vectoriales con -1 o 0
        if (intrinsicW <= 0 || intrinsicH <= 0) {
            super.onMeasure(widthMeasureSpec, heightMeasureSpec)
            return
        }

        val width = MeasureSpec.getSize(widthMeasureSpec)
        val imageRatio = intrinsicW.toFloat() / intrinsicH.toFloat()

        val height = if (width > 0) {
            (width / imageRatio).toInt()
        } else {
            intrinsicH
        }

        setMeasuredDimension(width, height)
    }
}
